frontend/src/metabase/admin/settings/components/widgets/ModelCachingScheduleWidget/CronExpressionInput.tsx

```diff
@@ -1,24 +1,72 @@
 import cx from "classnames";
 import type * as React from "react";
-import { useCallback, useState } from "react";
+import { useState } from "react";
 import { jt, msgid, ngettext, t } from "ttag";
 
 import TippyPopover from "metabase/components/Popover/TippyPopover";
 import ExternalLink from "metabase/core/components/ExternalLink";
 import FormS from "metabase/css/components/form.module.css";
 import CS from "metabase/css/core/index.css";
 import { validateCronExpression } from "metabase/lib/cron";
+import { Icon } from "metabase/ui";
 
-import {
-  CustomScheduleLabel,
-  ErrorMessage,
-  InfoIcon,
-  InputContainer,
-  PopoverContent,
-  PopoverText,
-  PopoverTitle,
-  StyledInput,
-} from "./CronExpressionInput.styled";
+import S from "./CronExpressionInput.module.css";
+import { CustomScheduleExplainer } from "./CustomScheduleExplainer";
+
+type CronExpressionInputProps = {
+  value: string;
+  onChange: (value: string) => void;
+  onBlurChange: (value: string) => void;
+};
+
+export function CronExpressionInput({
+  onChange,
+  onBlurChange,
+  value,
+}: CronExpressionInputProps) {
+  const [error, setError] = useState<string | null>(null);
+  const handleChange = (cronExpression: string) => {
+    // We don't allow to specify "seconds" and "year" components,
+    // "seconds" are mandatory for validation, so we're appending it at the beginning
+    let error = validateCronExpression(`0 ${cronExpression}`);
+    if (!error) {
+      error = validateExpressionComponents(cronExpression);
+    }
+    if (error) {
+      setError(error);
+    } else {
+      setError(null);
+    }
+    onChange(cronExpression);
+  };
+
+  const handleBlur = (cronExpression: string) => {
+    if (!error) {
+      // We don't allow to specify "seconds" and "year" components, but they're present in the value
+      // and we need to append them before sending a new value to the backend
+      onBlurChange(`0 ${cronExpression} *`);
+    }
+  };
+
+  return (
+    <>
+      <CustomScheduleInputHint />
+      <div className={S.inputContainer}>
+        <Input
+          value={value}
+          hasError={!!error}
+          onChange={handleChange}
+          onBlurChange={handleBlur}
+        />
+        <CronFormatPopover>
+          <Icon className={S.infoIcon} name="info" />
```

Comment:
@kdoh, should this "i" be inside a circle?

<img width="460" alt="image" src="https://github.com/user-attachments/assets/b3ecbff8-f26c-41b3-8385-d2771e1242bb" />

Reply:
FYI I switched to a `Tooltip` component, which may or may not influence how we style the 'i'

<img width="707" alt="Screenshot 2025-03-07 at 2 28 58 PM" src="https://github.com/user-attachments/assets/3a55680b-adda-4a69-aba1-67fcb2a951c6" />